web3.js/src/connection.ts

```diff
@@ -2554,17 +2601,33 @@ export class Connection {
     startSlot: number,
     endSlot: number,
   ): Promise<Array<TransactionSignature>> {
-    const unsafeRes = await this._rpcRequest(
-      'getConfirmedSignaturesForAddress',
-      [address.toBase58(), startSlot, endSlot],
-    );
-    const res = create(unsafeRes, GetConfirmedSignaturesForAddressRpcResult);
-    if ('error' in res) {
-      throw new Error(
-        'failed to get confirmed signatures for address: ' + res.error.message,
-      );
+    let options: any = {};
+
+    if (startSlot > 1) {
+      let start = startSlot - 1;
+      let startSignatures: Array<string> = [];
+      while (startSignatures.length == 0) {
+        const block = await this.getConfirmedBlockSignatures(start);
```

Comment:
If `start` is a skipped block, this method will throw

Reply:
@CriesofCarrots this is still an issue

Reply:
Sorry, I actually meant to ask you if you have recommendations for this, since the Connection methods don't distinguish between different kinds of errors. Is a string match the best we can do to distinguish skipped blocks (where we should continue the loop)?

Reply:
String match seems fine for now since we have no precedent for error codes yet. I think that can be pushed off until we adopt it across all of `Connection` methods

Reply:
Roger. https://github.com/solana-labs/solana/pull/16534/commits/917679ef92c80d1e00a1df8f77f8c6650fa7215a